### PR TITLE
(ios) Liquidity ads

### PR DESCRIPTION
**Pull request by robbiehanson**
<br/>

With a nice little animation:

https://github.com/ACINQ/phoenix/assets/304604/252a0552-4bf3-432e-9009-0c53265a9a3d

### CODE REVIEW

File: phoenix-ios/phoenix-ios/views/configuration/general/liquidity management/LiquidityAdsView.swift [hunk 1]
```diff
@@ -0,0 +1,1006 @@
...
+				If the bucket needs to be resized to allow for more water, \
+				that will require an on-chain operation with a mining fee.
+				""")
+				
+				Text(
+				"""
+				You may be able to reduce your fees by increasing your bucket size \
+				ahead of a series of incoming payments.
+				""")
+
```
Comment (dpad85) [inline phoenix-ios/phoenix-ios/views/configuration/general/liquidity management/LiquidityAdsView.swift:123]: There are several issues with this message:
- quite long (it's scary and people won't read it)
- takes a lot of screen space (on my phone I have to scroll down to reach the interesting part)
- the bucket analogy is kinda advanced and also already in the [FAQ](https://phoenix.acinq.co/faq#what-is-inbound-liquidity).

So the message should be as short as possible (something around "planning ahead to save money later"), with a link to the FAQ for users who want to know more. The FAQ is easier to maintain (shared between iOS and Android), and can be easily updated if we want to add more information.

**Review by dpad85 — CHANGES_REQUESTED** [review]
A few nitpicks:
- we should have a button to request liquidity in the Home screen, similar to the Android app. Without users will not be aware of this new feature. This is temporary, eventually this liquidity button should not be so prominent once we have reworked the onboarding UX.
- buttons navigating to the Request liquidity screen should not be visible if the user does not have any channels

File: phoenix-ios/phoenix-ios/views/configuration/general/liquidity management/LiquidityAdsView.swift [hunk 2]
```diff
@@ -0,0 +1,1006 @@
...
+				// User changed slider value while we were calculating the estimated fees.
+				// So the estimation is now invalid, and we can ignore it.
+				return
+			}
+			
+			if let pair = pair,
+				let fees: Lightning_kmpChannelCommand.CommitmentSpliceFees = pair.second
+			{
+				feeInfo = LiquidityFeeInfo(
+					params: LiquidityFeeParams(amount: amount, feerate: feePerKw, leaseRate: leaseRate),
```
[the remark this answers is not in the document]
Comment (robbiehanson) [reply phoenix-ios/phoenix-ios/views/configuration/general/liquidity management/LiquidityAdsView.swift]: Done in 4778664b

File: phoenix-ios/phoenix-ios/views/configuration/general/liquidity management/LiquidityAdsView.swift [hunk 3]
```diff
@@ -0,0 +1,1006 @@
...
+			
+			do {
+				pair = try await peer._estimateFeeForInboundLiquidity(
+					amount: amount,
+					targetFeerate: feePerKw,
+					leaseRate: leaseRate
+				)
+				
+				if pair == nil {
+					log.error("peer.estimateFeeForInboundLiquidity() == nil")
```
[the remark this answers is not in the document]
Comment (robbiehanson) [reply phoenix-ios/phoenix-ios/views/configuration/general/liquidity management/LiquidityAdsView.swift:749]: Done in c7d10c4d

File: phoenix-ios/phoenix-ios/views/configuration/general/liquidity management/LiquidityAdsView.swift [hunk 4]
```diff
@@ -0,0 +1,1006 @@
...
+			var result: Lightning_kmpChannelCommand.CommitmentSpliceResponse? = nil
+			do {
+				result = try await peer._requestInboundLiquidity(
+					amount: feeInfo.params.amount,
+					feerate: feeInfo.params.feerate,
+					leaseRate: feeInfo.params.leaseRate
+				)
+				
+				if result == nil {
+					log.error("peer.requestInboundLiquidity() == nil")
```
[the remark this answers is not in the document]
Comment (robbiehanson) [reply phoenix-ios/phoenix-ios/views/configuration/general/liquidity management/LiquidityAdsView.swift:803]: Done in c7d10c4d

**Comment by robbiehanson** [comment]
New minimal design:



Standard "help" button in upper-right corner, displays a help screen:



And a new "add liquidity" button on the home screen (only displayed if user has a channel):

**Review by dpad85 — APPROVED** [review]
🚀

**Pull request merged**